scripting/cache.go

```diff
@@ -0,0 +1,68 @@
+package scripting
+
+import (
+	"sync"
+
+	"github.com/mongodb/jasper"
+	"github.com/mongodb/jasper/options"
+	"github.com/pkg/errors"
+)
+
+// NewCache constructs a threadsafe HarnessCache instances.
+func NewCache() HarnessCache { return &cacheImpl{cache: make(map[string]Harness)} }
+
+type cacheImpl struct {
+	cache map[string]Harness
+	mutex sync.RWMutex
+}
+
+func (c *cacheImpl) Create(jpm jasper.Manager, opts options.ScriptingHarness) (Harness, error) {
+	c.mutex.Lock()
```

Comment:
Is this lock necessary if you're not modifying the cache? Or is this supposed to add the newly-created harness to the cache?

Reply:
good catch!